custom/icds_reports/static/js/directives/location-filter/location-filter.directive.js

```diff
@@ -270,7 +270,7 @@ function LocationFilterController($scope, $location, $uibModal, locationHierarch
                             }
                         });
 
-                        if (((["null", "undefined"].indexOf(userLocationId) === -1 && !haveAccessToAllLocations) || userLocationInSorted.length > 0) && !haveAccessToAllLocationsForLevel) {
+                        if (((["null", "undefined"].indexOf(userLocationId) === -1 && !haveAccessToAllLocationsForLevel) || userLocationInSorted.length > 0) && !haveAccessToAllLocations) {
```

Comment:
this type of long conditional can benefit from being extracted into a named function or variable to make it clearer what the statement is logically checking for.